aos/aos.py

```diff
@@ -111,7 +111,7 @@ def raw_request(self, method, uri, params, data, headers):
             raise AosAuthenticationError(
                 f"Authentication failed: {err_message(resp)}"
             )
-        elif resp.status_code >= 400:
+        elif resp.status_code >= 400 and resp.status_code != 404:
```

Comment:
Case with 404 response is already handled as expected by returning `None` in `raw_request_json()` method. When do you experience problems with 404 error response?

Reply:
pls check Traceback in #8

Reply:
Ok, I see where the problem is. HTTP 404 code must be handled separately, and removed from raw_request_json():

```
if resp.status_code == 401:
    raise AosAuthenticationError(
        f"Authentication failed: {err_message(resp)}"
    )
elif resp.status_code == 404 and method == "GET":
    return None
elif resp.status_code >= 400:
    raise AosAPIError(err_message(resp))
```

Reply:
@roman-verchikov 
```python
if resp.status_code == 401:
    raise AosAuthenticationError(
        f"Authentication failed: {err_message(resp)}"
    )
elif resp.status_code == 404 and method == "GET":
    return resp   <----- ! -> # as this needs go back to "raw_request_json" #
elif resp.status_code >= 400:
    raise AosAPIError(err_message(resp))
```
but it won't be that simple... in that situation we will add this to all raw_request -> which will cause that multiple tox tests failure (as direct get requests which points on not exist elements will also return now 404 instead of AosAPIError ).

We need somehow mark that this request initially comes from "repeat_until" and only for that kind of requests do not rise AosAPIError --or-- accept change of behavior and change tox scenarios so instead of AosAPIError we will get empty resp.status_code 404